### PR TITLE
Add explicit action inventory dump

### PR DESCRIPTION
**Pull request by mycontributiontothisscam**
Hi,
I've finally gotten around to packaging up some additions/modifications to stashie that I've used for a while that I believe other people would find useful.
Among the additions and modifications are:
1) Support for "veiled" and "incubator" keyword support.
2) Added "dumb drop"; that is, the ability to everything in your inventory (subject to ignored cells) whatever window is open.  I've used dump items into a vendor window and also to any individual open stash tab.  However, I've not tested it with open windows from other human players.
  a)  It is triggered by pressing <configured hotkey> + shift (release)
     That is, press both your configured hotkey and the shift button and, upon release of shift button, stashie will dump your inventory items not located on ignored cells into open the open window.
3) Miscellaneous speed improvements.
  a)  Stashie now orders the list of items to drop by stash number.  This mitigates "stash hopping" and essentially drops items into tabs in sequential order.

Thanks.

### CODE REVIEW

**Comment by zaafar** [comment]
Q: why do we need “incubator” keyword when we can simply do that via ‘ClassName^Incubator’?

File: StashieCore.cs [hunk 1]
```diff
@@ -306,140 +313,218 @@ private FilterResult CheckFilters(ItemData itemData)
...
-                    var dropdown = dropdownMenu.Children[index];
-                    for (var i = 0; i < dropdown.Children.Count; i++)
+                    Mouse.SetCursorPos(cursorPosPreMoving.X, cursorPosPreMoving.Y);
+                    if (Keyboard.IsKeyToggled(Settings.DropHotkey.Value))
                     {
-                        var child = dropdown.Children[i];
-                        Graphics.DrawBox(child.GetClientRect(), Color.DarkRed);
-                        var pos2 = new Vector2(child.GetClientRect().Center.X, child.GetClientRect().Top);
-                        Graphics.DrawText($"[{index}][{i}]", 20, pos2, Color.White, FontDrawFlags.Center);
+                        // This effectively "swallows" the configured hotkey making exclusive use of it...is that ok?  Should we just allow
```
Comment (zaafar) [inline StashieCore.cs]: swallow it is better.

**Comment by zaafar** [comment]
Q: The "dumb" dump to stash simply drops all items from inventory
Does this also work for the Trading Window??? If yes, then cool...wanted to do this.

File: StashieCore.cs (hunk 1 above)
Comment (mycontributiontothisscam) [reply StashieCore.cs]: Ok - would you mind editing that bit out and, if you don't mind, an short explanation why?
Collaborators should have write permission IIRC.  If  you need me to do it (or anything else) let me know.  Thanks.

**Comment by mycontributiontothisscam** [comment]
@zaafar 
Sorry I didn't see this earlier...
Q: why do we need “incubator” keyword when we can simply do that via ‘ClassName^Incubator’?
It's not needed - it's more a convenience thing.  For instance (and as an aside) emblems are treated as "MapFragment" (ie "Sacrifice at Dusk") via ClassName.  So one would need something like Name^<specific type>Emblem to properly differentiate them.

At some point, an argument can be made that stashie would be more accessible to more people if they didn't have to understand metadata qualifiers to arrive at what they want.
i.e.  incubator :incubator vs ClassName^Incubator :incubators
       EternalEmblem :emblems vs Name^Timeless Eternal Emblem :emblems

In other words, it's a trade-off of hiding complexity.  If you want, I can remove it (or you can) - I don't feel strongly one way or another.

**Comment by zaafar** [comment]
@mycontrobutiontothisscam sorry for not getting back to you. I was busy this weekend. I am going to commit some changes to your Pr and the. Merge it. I want this PR to be merged.

**Comment by zaafar** [comment]
@mycontributiontothisscam ^

**Comment by mycontributiontothisscam** [comment]
No problem.   BTW, I put the latest changes on the master branch as you may
have noticed.

Also,
1) Concerning whether the dump stash functionality works on trading windows
interacting with humans...I don't think that works ATM...DumpToOpenWindow
is allowed to progress when the vendor window is open
(GameController.Game.IngameState.UIRoot.Children[1].Children[74]) and when
the stash is open.  I presume the trading window isn't defined at UIRoot(1,
74) like the vendor window...but I haven't been able to test that aspect as
I don't have another account with which to find the right children.  If you
have that data handy, I could add it in.

2) Regarding the "Incubator" functionality - I planned on implementing a
sort of macro expansion (optionally recursive) keyword to obviate the need
for new support that would work off compositions of already defined
primitives.
For instance, you could write:
<FiltersConfig.txt>
.
Incubator@    ClassName^Incubator

Dump:   Incubator, ClassName=DivinationCard, <...>  :Dump Tab

I feel it offers a good compromise between filter config succinctness and
the continual need for code tweaks.
Thanks.
